src/CC/ShellCheck/ShellScript.hs

```diff
@@ -20,9 +22,24 @@ import           System.FilePath.Posix
 
 --------------------------------------------------------------------------------
 
+-- | List of shells the engine should be able to handle.
+validShells :: [BS.ByteString]
+validShells = ["sh", "ash", "dash", "bash", "ksh"]
+
+--------------------------------------------------------------------------------
+
+-- | List of valid shell file extensions.
+validShellExtensions :: [BS.ByteString]
+validShellExtensions = ("." <>) <$> validShells
+
+--------------------------------------------------------------------------------
+
 -- | Checks to see if file has correct extension.
 hasShellExtension :: FilePath -> Bool
-hasShellExtension path = takeExtension path == ".sh"
+hasShellExtension path = extension `elem` validShellExtensions
```

Comment:
@acook I think removed the duplicate applications of `takeExtension` and replaced it with a check that the extension is present in the list of valid shell extensions.